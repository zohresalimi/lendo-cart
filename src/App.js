import { Server } from "miragejs";

import myData from "./mockProducts.json";
import useAxios from "./hooks/useAxios";

new Server({
  routes() {
    this.namespace = "api";
    this.get("/products", () => {
      return myData;
    });
  },
});

function App() {
  const [{ data, isLoading, isError }, doFetch] = useAxios();
  console.log("data", data);
  console.log("isLoading", isLoading);
  console.log("isError", isError);
  console.log("doFetch", doFetch);
  return <div className="App">App</div>;
}

export default App;
