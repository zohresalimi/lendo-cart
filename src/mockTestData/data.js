import React, { Children, useReducer } from "react";
import { BrowserRouter as Router } from "react-router-dom";

import AppContext from "../store/context";
import reducer from "../store/reducer";

export const getTestStore = () => {
  return {
    currentProduct: {
      id: 1,
      selectedFeatures: {
        color: "whith",
        power: 6.5,
      },
    },
    products: {
      byColor: {
        white: {
          1: { power: [6.5, 9.5], remaining: 3 },
          2: { power: [6.5, 9.5], remaining: 3 },
        },
        red: {
          1: { power: [6.5, 9.5], remaining: 7 },
          2: { power: [6.5, 9.5], remaining: 7 },
        },
      },
      byId: {
        1: {
          id: 1,
          name: "Philips hue bulb",
          brand: "Philips",
          price: "500",
          available: true,
          weight: 0.2,
          options: [
            {
              color: "white",
              power: [6.5, 9.5],
              quantity: 3,
            },
            {
              color: "red",
              power: [6.5, 9.5],
              quantity: 7,
            },
          ],
        },
        2: {
          id: 2,
          name: "TrÃ¥dfria Lampor",
          brand: "IKEA",
          price: "300",
          available: true,
          weight: 0.2,
          options: [
            {
              color: "white",
              power: [6.5, 9.5],
              quantity: 3,
            },
            {
              color: "red",
              power: [6.5, 9.5],
              quantity: 7,
            },
            {
              color: "green",
              power: [6.5],
              quantity: 0,
            },
          ],
        },
      },
    },
    shoppingCart: [
      { id: 1, quantity: 1, selectedFeatures: { color: "white", power: 6.5 } },
    ],
  };
};

export const WithProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, getTestStore());

  return (
    <AppContext.Provider
      value={{
        state,
        dispatch: props.mockDispatch || dispatch,
      }}
    >
      <Router>{props.children}</Router>
    </AppContext.Provider>
  );
};
