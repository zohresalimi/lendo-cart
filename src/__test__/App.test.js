import { act, render, screen } from "@testing-library/react";
import App from "../App";

import { WithProvider } from "../mockTestData/data";

jest.mock("../components/NavBar", () => () => <div>NavBar Component</div>);
jest.mock("../AppSwitch", () => () => <div>AppSwitch Component</div>);
async function renderWrapper(props) {
  let component;
  act(() => {
    component = render(
      <WithProvider>
        <App />
      </WithProvider>
    );
  });
  return component;
}

describe("App Component Testing", () => {
  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });
});
