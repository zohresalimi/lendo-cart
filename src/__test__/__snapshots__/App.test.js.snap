// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Testing take snapshot 1`] = `
<div>
  NavBar Component
</div>
`;
